vault/resource_gcp_secret_static_account.go

```diff
@@ -0,0 +1,263 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	gcpSecretStaticAccountBackendFromPathRegex = regexp.MustCompile("^(.+)/static-account/.+$")
+	gcpSecretStaticAccountNameFromPathRegex    = regexp.MustCompile("^.+/static-account/(.+)$")
+)
+
+func gcpSecretStaticAccountResource() *schema.Resource {
+	return &schema.Resource{
+		Create: gcpSecretStaticAccountCreate,
+		Read:   gcpSecretStaticAccountRead,
+		Update: gcpSecretStaticAccountUpdate,
+		Delete: gcpSecretStaticAccountDelete,
+		Exists: gcpSecretStaticAccountExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Path where the GCP secrets engine is mounted.",
+				ForceNew:    true,
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"static_account": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the Static Account to create",
+				ForceNew:    true,
+			},
+			"secret_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Type of secret generated for this static account. Defaults to `access_token`. Accepted values: `access_token`, `service_account_key`",
+			},
+			"service_account_email": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Email of the GCP service account.",
+			},
+			"token_scopes": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "List of OAuth scopes to assign to `access_token` secrets generated under this role set (`access_token` role sets only) ",
+			},
+			"binding": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      gcpSecretBindingHash,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Resource name",
+						},
+						"roles": {
+							Type:        schema.TypeSet,
+							Required:    true,
+							Description: "List of roles to apply to the resource",
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"service_account_project": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Project of the GCP Service Account managed by this static account",
+			},
+		},
```

Comment:
Do you know if the [CustomizeDiff](https://github.com/hashicorp/terraform-provider-vault/blob/master/vault/resource_gcp_secret_roleset.go#L105-L114) used for rolesets is relevant static accounts?

Reply:
Got it. Thanks for the explanation!